### DISCUSSION TITLE
[@types/humps] Doesn't camelize values that are potentially null/undefined

### DISCUSSION THREAD

**Discussion by markoj3s** [DefinitelyTyped/DefinitelyTyped#65134] [Issues with a @types package] [answered]
Hi all~!

I am facing an issue whenever a value may be null/undefined.

As an example, let's say I have below random piece of code;
```
const fooBar: FooBar = { f_oo: { b_ar: 1 } };
const camelizedFooBar = camelizeKeys<FooBar>(fooBar);
camelizedFooBar.fOo.bAr // Attempt to access bAr
```

If FooBar type alias is defined as
```
type FooBar = {
  f_oo: { b_ar: number };
};
```
typescript would be happy and recognize the existence of `fOo` and `bAr`.

However, if FooBar is defined as
```
type FooBar = {
  f_oo?: { b_ar: number };
};
```
or
```
type FooBar = {
  f_oo: { b_ar: number } | null;
};
```
then typescript would only recognize `fOo` and `b_ar` but not `bAr`. (so it will camelize up to the undefined/null value)

[a comment by an automated account is not shown]

**Comment by nikeee** [comment]
It seems that because the property is nullable, this won't branch into transformation:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/369784b434c20f73710cb8c42eccea78bed8d4d6/types/humps/index.d.ts#L13-L21

We should probably allow null/undefined as a base value and apply that the the transformed property, so it seems like a bug to me. I currently don't have the resources to fix this. If you want to, go ahead! (cc @peterblazejewicz)

**Comment by dominikjasek** [comment, marked as answer]
Fixed here: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/65884

**Comment by markoj3s** [comment]
Thank you for your reply @nikeee and sorry for my late reply, I did not notice it.
Thanks a lot @dominikjasek for the fix~!
